### PR TITLE
SpecDec Bench: April Update

### PR DESCRIPTION
**Pull request by IzzyPutterman**
### What does this PR do?

Type of change: ? 

### Usage

```python
# Add a code snippet demonstrating how to use this
```

### Before your PR is "*Ready for review*"

Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md) and your commits are signed (`git commit -s -S`).

Make sure you read and follow the [Security Best Practices](https://github.com/NVIDIA/Model-Optimizer/blob/main/SECURITY.md#security-coding-practices-for-contributors) (e.g. avoiding hardcoded `trust_remote_code=True`, `torch.load(..., weights_only=False)`, `pickle`, etc.).

- Is this change backward compatible?: ✅ / ❌ / N/A 
- If you copied code from any other sources or added a new PIP dependency, did you follow guidance in `CONTRIBUTING.md`: ✅ / ❌ / N/A 
- Did you write any new necessary tests?: ✅ / ❌ / N/A 
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: ✅ / ❌ / N/A 

## Summary by CodeRabbit

* **New Features**
  * YAML sweep support with per-run overrides, new CLI flags, per-run output directories, and a sweep example config
  * Added HumanEval dataset and dataset-category filtering
  * New Thinking-Acceptance metric

* **Improvements**
  * Metrics now report acceptance-lengths, include raw choices and output token IDs; removed output_logits from outputs
  * Support for pre-built conversation requests, per-run temperature/output-length control, bounded concurrency, progress reporting, and environment dump

* **Documentation**
  * Updated installation image tags and detailed sweep/usage guide

### CODE REVIEW

**Comment by kevalmorabia97** [comment]
@IzzyPutterman 

1. can you update PR title and description with what's changed
2. Address Coderabbit's Critical and Major concerns?
3. Fix merge conflicts

File: examples/specdec_bench/specdec_bench/models/base.py [hunk 1]
```diff
@@ -13,19 +13,55 @@
...
+        beam_state.final_tokens = cumulative_tokens
+        beam_state.token_count = new_count
+        self.token_times.append(timestamp)
+        return True
+
+    def output_ids(self):
+        return [list(self._beam_states[i].final_tokens) for i in sorted(self._beam_states)]
+
+    def chunk_lengths(self):
+        return [list(self._beam_states[i].chunk_lengths) for i in sorted(self._beam_states)]
```
[the remark this answers is not in the document]
Comment (IzzyPutterman) [reply examples/specdec_bench/specdec_bench/models/base.py:67]: Beams are always 1 in this repo atm

File: examples/specdec_bench/specdec_bench/models/vllm.py [hunk 2]
```diff
@@ -31,122 +31,102 @@
...
     def stop(self):
-        try:
-            self.loop.run_until_complete(self.model.shutdown())
-            self.loop.close()
-        except Exception:
-            pass
+        import contextlib
+
+        with contextlib.suppress(Exception):
+            asyncio.run(self.model.shutdown())
```
[the remark this answers is not in the document]
Comment (IzzyPutterman) [reply examples/specdec_bench/specdec_bench/models/vllm.py:132]: This happens at the end to clean the setup, not worried about any issues

File: examples/specdec_bench/specdec_bench/utils.py [hunk 3]
```diff
@@ -14,15 +14,21 @@
...
+import os
+import sys
 
 from transformers import AutoTokenizer
 
+_SENSITIVE_SUBSTRINGS = ("token", "key", "secret", "password")
 
-def get_tokenizer(path, trust_remote_code=False):
+
+def get_tokenizer(path, trust_remote_code=True):
```
Comment (kevalmorabia97) [inline examples/specdec_bench/specdec_bench/utils.py]:
```suggestion
def get_tokenizer(path, trust_remote_code=False):
```

File: examples/specdec_bench/specdec_bench/metrics/specbench.py [hunk 4]
```diff
@@ -42,29 +42,33 @@ def __init__(self, requests):
...
         for request_id, request in enumerate(self.requests):
-            turns = self.prompt_ar[request_id].values()
+            turns = prompt_ar[request_id].values()
             assert len(turns) == len(request.turns), (
                 f"Number of turns {len(turns)} does not match number of turns in request {len(request.turns)}"
             )
-            self.out["Request_AR"][request.question_id] = mean(list(chain(*turns)))
+            self.out["Request_AL"][request.question_id] = mean(list(chain(*turns)))
             for turn in turns:
                 self._get_lengths(turn, lengths)
```
[the remark this answers is not in the document]
Comment (IzzyPutterman) [reply examples/specdec_bench/specdec_bench/metrics/specbench.py:58]: This should fail
